pynamodb/tests/integration/test_transaction_integration.py

```diff
@@ -0,0 +1,247 @@
+import os
+import uuid
+from datetime import datetime
+
+import pytest
+
+from pynamodb.connection import Connection
+from pynamodb.exceptions import PutError, DoesNotExist
+
+from pynamodb.attributes import NumberAttribute, UnicodeAttribute, UTCDateTimeAttribute, BooleanAttribute
+from pynamodb.connection.transactions import TransactGet, TransactWrite
+
+from pynamodb.models import Model
+
+IDEMPOTENT_PARAMETER_MISMATCH = 'IdempotentParameterMismatchException'
+PROVISIONED_THROUGHPUT_EXCEEDED = 'ProvisionedThroughputExceededException'
+RESOURCE_NOT_FOUND = 'ResourceNotFoundException'
+TRANSACTION_CANCELLED = 'TransactionCanceledException'
+TRANSACTION_IN_PROGRESS = 'TransactionInProgressException'
+
+
+DDB_URL = os.getenv('PYNAMODB_INTEGRATION_TEST_DDB_URL', 'http://localhost:8000')
+
+
+class User(Model):
+    class Meta:
+        host = DDB_URL
+        region = 'us-east-1'
+        table_name = 'user'
+
+    user_id = NumberAttribute(hash_key=True)
+
+
+class BankStatement(Model):
+
+    class Meta:
+        host = DDB_URL
+        region = 'us-east-1'
+        table_name = 'statement'
+
+    user_id = NumberAttribute(hash_key=True)
+    balance = NumberAttribute(default=0)
+    active = BooleanAttribute(default=True)
+
+
+class LineItem(Model):
+
+    class Meta:
+        host = DDB_URL
+        region = 'us-east-1'
+        table_name = 'line-item'
+
+    user_id = NumberAttribute(hash_key=True)
+    created_at = UTCDateTimeAttribute(range_key=True, default=datetime.now())
+    amount = NumberAttribute()
+    currency = UnicodeAttribute()
+
+
+class DifferentRegion(Model):
+
+    class Meta:
+        host = DDB_URL
+        region = 'us-east-2'
+        table_name = 'different-region'
+
+    entry_index = NumberAttribute(hash_key=True)
+
+
+TEST_MODELS = [
+    BankStatement,
+    DifferentRegion,
+    LineItem,
+    User,
+]
+
+
+CONNECTION = Connection(host=DDB_URL)
+
+
+class TestTransactionIntegration:
```

Comment:
Can we have the tests as functions rather than class methods? This is how modern pytest tests should be written. For setup and teardown that's per-module, do:
```
@pytest.fixture(scope='module', autouse=True)
def create_tables(ddb_url):
    for m in TEST_MODELS:
        m.Meta.host = ddb_url
        m.create_table(
            read_capacity_units=10,
            write_capacity_units=10,
            wait=True
        )
    yield
    for m in TEST_MODELS:
        if m.exists():
            m.delete_table()
```

btw, 'create_table' already checks for 'exists'

Reply:
sure :0